### PR TITLE
HHH-17797 support query hint database indexes

### PR DESCRIPTION
**Pull request by jeonjonghyeok**
https://hibernate.atlassian.net/browse/HHH-17797

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/jpa/HibernateHints.java [hunk 1]
```diff
@@ -223,4 +223,11 @@ public interface HibernateHints {
 	 * @since 6.3
 	 */
 	String HINT_QUERY_PLAN_CACHEABLE = "hibernate.query.plan.cacheable";
+
+	/**
+	 * Hint to enable adding indexes to database queries
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/jpa/HibernateHints.java:228]:
```suggestion
	 * Hint for specifying a query hint to be appended to the 
	 * SQL statement sent to the database.
```

Comment (jeonjonghyeok) [reply hibernate-core/src/main/java/org/hibernate/jpa/HibernateHints.java]: Thanks for the suggestion😃

**Review by jeonjonghyeok — COMMENTED** [review]
Thanks for the suggestion~!

File: hibernate-core/src/main/java/org/hibernate/query/spi/AbstractCommonQueryContract.java [hunk 2]
```diff
@@ -547,6 +551,9 @@ protected void applyFlushModeHint(FlushMode flushMode) {
 		setHibernateFlushMode( flushMode );
 	}
 
+	protected void applyDatabaseHint(String hint) {
+		getQueryOptions().addDatabaseHint(hint);
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/spi/AbstractCommonQueryContract.java:555]:
```suggestion
		getQueryOptions().addDatabaseHint( hint );
```

**Review by beikov — APPROVED** [review]
Please squash to a single commit

File: hibernate-core/src/main/java/org/hibernate/query/spi/AbstractCommonQueryContract.java (hunk 2 above)
Comment (jeonjonghyeok) [reply hibernate-core/src/main/java/org/hibernate/query/spi/AbstractCommonQueryContract.java]: done thanks 😀

**Pull request merged**